validation/pipeline/qase/reporter/main.go

```diff
@@ -46,12 +49,29 @@ func main() {
 
 		runID, err := strconv.ParseInt(runIDEnvVar, 10, 64)
 		if err != nil {
-			logrus.Fatalf("error reporting converting string to int64: %v", err)
+			logrus.Fatalf("error converting run ID string to int64:: %v", err)
 		}
 
-		err = reportTestQases(client, runID)
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+		defer cancel()
+
+		err = wait.PollUntilContextTimeout(ctx, 5*time.Second, 10*time.Minute, true, func(ctx context.Context) (bool, error) {
+			statusCode, err := reportTestQases(client, runID)
+			if err == nil {
```

Comment:
Can err be nil but statusCode also be `http.StatusTooManyRequests`?

Reply:
I don't believe so in this use case -- it appears that `"go.qase.io/client"` interprets `err != nil` as `status code is not 2xx` -- so `429` would render as an error and this implementation should be correct 👍